### PR TITLE
add rainbow tank

### LINKED ISSUE #43: Create a rainbow tank
Sometimes, a random rainbow tank might spawn in the game. I don't know the exact odds, but they are rare. 

- [x] Add a new boolean in the Enemy class called "rainbow"
- [x] Also add this to the constructor
- [x] In the spawn_enemies() method, add the small possibility of a tank to become a rainbow tank (like 1/20)
- [x] In the draw() method in the Enemy class, change the color of the enemies if they are a rainbow tank (it looks like you can do this with the [tint method](https://www.processing.org/reference/tint_.html))

This is it for now, we will later add the feature were the player gets a superpower if they destroy the rainbow tank.

### PR DESCRIPTION
**Pull request by max2032**

### CODE REVIEW

File: tank_battalion/enemy.pde [hunk 1]
```diff
@@ -9,13 +9,19 @@ class Enemy {
...
+  color color_red =color(255,0,0);
 
   final String SPRITES_FOLDER = "../assets/sprites/";
   PImage enemy_up, enemy_down, enemy_left, enemy_right, actual_image;
   
-  Enemy(int xpos, int ypos){
+  Enemy(int xpos,boolean Rainboww, int ypos){
     x = xpos;
     y = ypos;
+    rainbow = true;
```
Comment (robkruger) [inline tank_battalion/enemy.pde]: You forgot to set this back to `rainbow = Rainboww;`. With this code every tank will be a rainbow tank.

File: tank_battalion/game_state.pde [hunk 2]
```diff
@@ -168,7 +174,7 @@ class GameState extends State
     
     for(int i = 0; i < n_lives; i++)
     {
-      image(tank_image, width - 300 + i * 100, 300, 64, 64);  
+      image(tank_image, width - 340 + i * 64, 600, 64, 64);
```
Comment (robkruger) [inline tank_battalion/game_state.pde:198]: Unrelated to this issue, but it's fine.

**Review by robkruger — DISMISSED** [review]
If you change `rainbow = true` to `rainbow = Rainboww` then it can be merged with `develop` in my opinion. Indention and variable names could be better, but it's fine for now. 👍

File: tank_battalion/game_state.pde (hunk 2 above)
Comment (rijkvp) [reply tank_battalion/game_state.pde:198]: That was actually from his previous issue which he forgot to commit. :wink: (I helped him)

**Review by rijkvp — CHANGES_REQUESTED** [review]
Yes I agree with @robkruger. Please fix the constructor of the enemy class.

**Review by robkruger — CHANGES_REQUESTED** [review]
Please change the things I said in my previous comment

**Comment by rijkvp** [comment]
@max2032, Nice you fixed the issue! But can you please rename `Rainbow` to `is_rainbow` the name is really confusing (for me).

**Comment by rijkvp** [comment]
@max2032, I fixed it for you (cuz I was bored). Would you please have a look at commit 72a2d00ddb9e63ef694a9a138d93fb6e673a045b?

**Comment by max2032** [comment]
Finally nice :D

Op di 6 okt. 2020 om 17:37 schreef Rijk van Putten <notifications@github.com

**Pull request merged**